xarray/tests/test_weighted.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+
+import numpy as np
+
+import xarray as xr
+from xarray import (
+    DataArray,)
+
+from xarray.tests import assert_equal, raises_regex
+
+
+def test_weigted_non_DataArray_weights():
+
+    da = DataArray([1, 2])
+    with raises_regex(AssertionError, "'weights' must be a DataArray"):
+        da.weighted([1, 2])
+
+
+@pytest.mark.parametrize('weights', ([1, 2], [np.nan, 2], [np.nan, np.nan]))
+def test_weighted_weights_nan_replaced(weights):
+    # make sure nans are removed from weights
+
+    da = DataArray([1, 2])
+
+    expected = DataArray(weights).fillna(0.)
+    result = da.weighted(DataArray(weights)).weights
+
+    assert_equal(expected, result)
+
+
+@pytest.mark.parametrize(('weights', 'expected'), (([1, 2], 3),
+                                                   ([0, 2], 2),
+                                                   ([0, 0], np.nan),
+                                                   ([-1, 1], np.nan)))
+def test_weigted_sum_of_weights_no_nan(weights, expected):
+
+    da = DataArray([1, 2])
+    weights = DataArray(weights)
+    result = da.weighted(weights).sum_of_weights()
+
+    expected = DataArray(expected)
+
+    assert_equal(expected, result)
+
+
+@pytest.mark.parametrize(('weights', 'expected'), (([1, 2], 2),
+                                                   ([0, 2], 2),
+                                                   ([0, 0], np.nan),
+                                                   ([-1, 1], 1)))
+def test_weigted_sum_of_weights_nan(weights, expected):
```

Comment:
Can we test with a weight with more than one dimensional?

Reply:
I still need to do this....

Reply:
done